circuits/cpp/barretenberg/ts/bin-test/test.js

```diff
@@ -0,0 +1,151 @@
+const expectedBarretenbergWasmLocation = path.join(scriptDirectory, '../dest/barretenberg.wasm');
+const expectedBarretenbergWasmThreadsLocation = path.join(scriptDirectory, '../dest/barretenberg-threads.wasm');
+const possibleBarretenbergWasmLocation = path.join(scriptDirectory, '../../cpp/build-wasm/bin/barretenberg.wasm');
+const possibleBarretenbergWasmThreadsLocation = path.join(scriptDirectory, '../../cpp/build-wasm-threads/bin/barretenberg.wasm');
+
+// Skips warnings, ie. experimental features
+process.removeAllListeners('warning');
+
+function getScriptDirectory() {
+  const moduleURL = import.meta.url;
+  const modulePath = fileURLToPath(moduleURL);
+  const scriptDirectory = path.dirname(modulePath);
+
+  return scriptDirectory
+}
+
+function copyBBWASMIfNotPresent(expectedLocation, possibleBuildLocation) {
+
+  if (!fs.existsSync(expectedLocation)) {    
+    try {
+      fs.copyFileSync(possibleBuildLocation, expectedLocation);
+      console.log(`${possibleBuildLocation} copied to ${expectedLocation}`);
+    } catch (error) {
+      console.error(`Error copying file ${possibleBuildLocation} to ${expectedLocation}:`, error);
+      console.error('Have you build with `yarn build` ?');
+      throw error;
+    }
+  }
+
+}
+
+function throwMissingFile(filePath, message) {
+
+  const fileExists = fs.existsSync(filePath);
+  if (!fileExists) {
+    console.error(`File does not exist at ${filePath}, ${message}.`); 
+    throw Error(`File does not exist at ${filePath}, ${message}.`);
+  }
+
+}
+
+function checkArtifactsPresent() {
+
+  copyBBWASMIfNotPresent(expectedBarretenbergWasmLocation, possibleBarretenbergWasmLocation);
+  copyBBWASMIfNotPresent(expectedBarretenbergWasmThreadsLocation, possibleBarretenbergWasmThreadsLocation);
+  
+  throwMissingFile(executionScript, "execute 'yarn build:ts:node' first");
+
+  throwMissingFile(compiledNoirProgramLocation, "execute 'nargo compile' first");
+
+  throwMissingFile(solvedWitnessLocation, "execute 'nargo execute' first");
+
+}
+
+describe('Test bb subcommands', () => {
+
+  before(checkArtifactsPresent);
+
+  it('get gate count using `bb gates` command', (done) => {
+    
+    const expectedOutputTemplate = /^gates: \d+$/;
+    const executionDirectory = getScriptDirectory();
+    const command = `${executionScript} gates`;
+
+    exec(command, { cwd: executionDirectory }, function(error, stdout, stderr) {
+      if (error) {
+        done(error);
+        return;
+      }
+
+      // Assert that the stdout matches the expected output template
+      assert.match(stdout.trim(), expectedOutputTemplate);
+
+      done();
+    });
+  });
+
+  it('creates proof with `bb prove -o proof` command', (done) => {
+    
+    const expectedOutputTemplate = /^proof written to: proof$/;
+    const executionDirectory = getScriptDirectory();
+    const command = `${executionScript} prove -o proof`;
+
+    exec(command, { cwd: executionDirectory }, function(error, stdout, stderr) {
```

Comment:
Seems like the exec commands of the tests can be sent over to a function that receives the excutionDirectory as parameter since they seem to be repeated across all tests. The only difference between the tests seem to be the expectedOuputTemplate and the command (executionDirectory remains the same?)

Reply:
efbddf8: fix: test execution